webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,234 @@
+    const fetchResourcePools = async () => {
+      setResourcePools( await getResourcePools({}));
+    };
+    fetchResourcePools();
+  }, []);
+
+  useEffect(()=> {
+    const fetchConfig = async (values: NotebookConfig) => {
+      if(showFullConfig) {
+        const config = await previewNotebook(
+          values.slots,
+          values.template,
+          values.name,
+          values.pool,
+        );
+        form.setFieldsValue({ config: yaml.dump(config) });
+      }
+    };
+    fetchConfig(form.getFieldsValue(true));
+  }, [ showFullConfig, form ]);
+
+  const storeConfig = useCallback((_, values: NotebookConfig) => {
+    delete values.name;
+    storage.set(STORAGE_KEY,values);
+  }, [ storage ]);
+
+  const handleSecondary = useCallback(() => {
+    if (showFullConfig) {
+      setShowFullConfig(show => !show);
+    } else {
+      form.validateFields().then(() => setShowFullConfig(show => !show)).catch();
+    }
+  },[ form, showFullConfig ]);
+
+  const handleCreateEnvironment = useCallback(
+    (values) =>{
+      if (showFullConfig) {
+        launchNotebook(yaml.load(form.getFieldValue('config')) as RawJson);
+      } else {
+        launchNotebook(
+          undefined,
+          values.resourceType === 'GPU'? values.slots : 0,
+          values.template === ''? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      onLaunch();
+    },
+    [ showFullConfig, form, onLaunch ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((e) => {
+    if (e === '') {
+      setResourceTypeOptions([ { id:'CPU', label:'CPU' }, { id:'GPU', label:'GPU' } ]);
+    } else {
+      const pool = resourcePools.find(pool => pool.name === e);
+      if (pool){
+        const options = [];
+        if (pool.cpuContainerCapacityPerAgent > 0) {
+          options.push({ id:'CPU', label:'CPU' });
+        }
+        if (pool.slotsPerAgent && pool.slotsPerAgent > 0) {
+          options.push({ id:'GPU', label:'GPU' });
+        }
+        setResourceTypeOptions(options);
+        form.setFieldsValue({ type:undefined });
+        setResourceType(undefined);
+      }
+    }
+  },[ resourcePools, form ]);
+
+  const handleTypeUpdate = useCallback((e) => {
+    setResourceType(e);
+  },[]);
+
+  return <Modal
+    footer={<>
+      <Button onClick={handleSecondary}>{showFullConfig ? 'Edit Form' : 'Edit Full Config'}</Button>
+      <Button
+        type="primary"
+        onClick={() => {
+          form.validateFields().then(values => {
+            handleCreateEnvironment(values);
+          }).catch();
+        }
+        }>Launch</Button>
+    </>}
+    title='Launch JupyterLab'
+    visible={visible}
+    {...props}>
+    {showFullConfig?
+      <Form form={form}>
+        <div style={{
+          backgroundColor:'rgb(230,230,230)',
+          border:'1px solid rgb(200,200,200)',
+          marginBottom: '4px',
+          padding: 2,
+          textAlign:'center',
+        }}
```

Comment:
**suggestion:** create a `NotebookModal.module.scss` file to extract and isolate styles and use CSS theme vars wherever possible. So it would be something like:
`NotebookModal.module.scss`
```
.note {
  background-color: var(--theme-colors-monochrome-16),
  ...,
}
```
`NotebookModal.tsx`
```
import css from './NotebookModal.module.scss';
...
<div className={css.note}>
  ...
</div>
```